### ISSUE TITLE
BUG: Paper Case got saved in an invalid state somehow

### ISSUE THREAD

**Issue by mmarcotte** [ustaxcourt/ef-cms#7937]
**Describe the Bug**

DocketNumber `10168-21` was in an invalid state in production and a recent glue job. 

In order to resolve it, we had to add a `secondaryName` to the petitioner in the array. The name of the petitioner was also incorrect.

Changed:
- `name`: `Estate of Michael I. Miller, Deceased`
- `secondaryName`: `Barbara Miller, Executrix`

Trying down log information about when this happened. @mmarcotte 

**Business Impact/Reason for Severity**

Invalid case caused the case to 404 when searching for case.

**In which environment did you see this bug?**

Prod and Mig

**Who were you logged in as?**

Any

**What were you doing when you discovered this bug? (Using the application, demoing, smoke tests, testing other functionality, etc.)**

Performing a migration on `mig`

**To Reproduce**

**Expected Behavior**

Cases should never get saved in an invalid state

**Actual Behavior**

Case was stored with missing `secondaryName` and incorrect `name`.

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Cause of Bug, If Known**

**Process for Logging a Bug:**
* Complete the above information
* Add a severity tag (Critical, High Severity, Medium Severity or Low Severity). See below for priority definition. 

**Severity Definition:**
* Critical Defect
Blocks entire system's or module’s functionality
No workarounds available
Testing cannot proceed further without bug being fixed.

* High-severity Defect
Affects key functionality of an application
There's a workaround, but not obvious or easy
App behaves in a way that is strongly different from the one stated in the requirements

* Medium-severity Defect
A minor function does not behave in a way stated in the requirements.
Workaround is available and easy

* Low-severity Defect
Mostly related to an application’s UI
Doesn't need a workaround, because it doesn't impact functionality

## Definition of Done (Updated 4-14-21)
**Product Owner**
 - [ ]  Bug fix has been validated in the Court's test environment

**Engineering**
 - [ ] Automated test scripts have been written
 - [ ] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [ ] Verify that language for docket record for internal users and external users is identical
 - [ ] New screens have been added to pa11y scripts
 - [ ] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [ ] READMEs, other appropriate docs, JSDocs and swagger/APIs fully updated
 - [ ] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [ ] Interactors should validate entities before calling persistence methods
 - [ ] Code refactored for clarity and to remove any known technical debt
 - [ ] Deployed to the Court's test environment

**Comment by TomElliottFlexion** [comment]
- PARTY_TYPE of estate
- secondaryName and title were undefined, so its computed additionalName was an empty string -- which joi throws a validation error for unless you explicitly specify it to allow empty strings.
- somehow an internal user was allowed to create this case (since it's a paper case) without a secondaryName, since that's required in the validation. odd. -Jojo

**Comment by sutt0n** [comment]
After attempting to reproduce this in multiple ways, and even modifying the request, we're unable to reproduce. Both frontend and backend validation thwart the attempt of passing a blank `secondaryName` and since modification deltas are not logged, it's very difficult (if not impossible) to determine the exact cause of this.

Marking as Unable to Reproduce.

**Comment by mmarcotte** [comment]
Just to add a little more information about how we tried to troubleshoot this. 

We restored a DDB Backup to a time before we corrected the issue. And we observed that the last time the record was updated was at the precise second that a Practitioner was removed from the case. It appears that API call somehow triggered the errant update. However, when we tried to replicate it with these steps, we continued to be unable to reproduce the problem. 

Perhaps this odd behavior was caused by an older version of DDB. And this ticket aims to upgrade that: https://github.com/flexion/ef-cms/issues/8485
